synapse/rest/client/v1/events.py

```diff
@@ -81,7 +81,7 @@ def __init__(self, hs):
         self._event_serializer = hs.get_event_client_serializer()
 
     async def on_GET(self, request, event_id):
-        requester = await self.auth.get_user_by_req(request)
+        requester = await self.auth.get_user_by_req(request, allow_guest=True)
```

Comment:
this endpoint is deprecated, and nothing should be using it. See https://matrix.org/docs/spec/client_server/r0.6.1#deprecated-get-matrix-client-r0-events-eventid.